### PR TITLE
Bump minimum deployment target to iOS 16

### PR DESCRIPTION
**Pull request by kean**
## To test 

n/a

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages in Xcode.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/WordPressShareExtension/ShareExtensionEditorViewController.swift [hunk 1]
```diff
@@ -981,9 +981,11 @@ extension ShareExtensionEditorViewController {
         alertController.popoverPresentationController?.sourceView = richTextView
         alertController.popoverPresentationController?.sourceRect = CGRect(origin: position, size: CGSize(width: 1, height: 1))
         alertController.popoverPresentationController?.permittedArrowDirections = .any
-        present(alertController, animated: true, completion: { () in
-            UIMenuController.shared.hideMenu()
-        })
+        present(alertController, animated: true) {
+            for interaction in self.richTextView.interactions {
```
Comment (kean) [inline WordPress/WordPressShareExtension/ShareExtensionEditorViewController.swift:985]: This code hides the menu that occasionally shows up when tapping on media, which still works with the new code:

https://github.com/user-attachments/assets/c47281d2-4c8f-414d-acee-ddc41638bc8b

**Comment by kean** [comment]
@mokagio, would you mind looking into it?

Last time I bumped the deployment target I simply had to update `IPHONEOS_DEPLOYMENT_TARGET` in `Common.xcconfig`, but this time I'm getting the following errors:

```
<head></head>
❌ /opt/ci/builds/builder/automattic/wordpress-ios/DerivedData/Build/Products/Debug-iphonesimulator/XCFrameworkIntermediates/Gutenberg/Gutenberg.framework/Modules/Gutenberg.swiftmodule/arm64-apple-ios-simulator.private.swiftinterface:5:8: compiling for iOS 15.0, but module 'Aztec' has a minimum deployment target of iOS 16.0: /opt/ci/builds/builder/automattic/wordpress-ios/DerivedData/Build/Products/Debug-iphonesimulator/WordPress-Aztec-iOS/Aztec.framework/Modules/Aztec.swiftmodule/arm64-apple-ios-simulator.swiftmodule
--
  | import Aztec
```

**Comment by crazytonyli** [comment]
@kean The error is probably caused by this configuration in Podfile. Not sure if it's still needed.

https://github.com/wordpress-mobile/WordPress-iOS/blob/b347b02ff9b70f1803298f72325ce2cd67cfec83/Podfile#L154-L166

**Comment by kean** [comment]
> @kean The error is probably caused by this configuration in Podfile. Not sure if it's still needed.

Thanks for the suggestions. I removed it, and looks like it's now working.

**Review by crazytonyli — APPROVED** [review]

**Review by mokagio — APPROVED** [review]

**Pull request merged**